lib/jekyll/document.rb

```diff
@@ -262,8 +284,12 @@ def published?
     def read(opts = {})
       Jekyll.logger.debug "Reading:", relative_path
 
-      if yaml_file?
+      if yaml_file? || json_file?
```

Comment:
maybe combine this with the `csv_file?` as a `data_file?` method. Then check for if its a `csv_file?` and process it accordingly.